android/src/main/java/com/github/amarcruz/rntextsize/RNTextSizeModule.java

```diff
@@ -326,23 +343,42 @@ private void flatHeightsInner(@Nullable final ReadableMap specs, final Promise p
                 layout = buildStaticLayout(conf, includeFontPadding, sb, textPaint, (int) width);
                 heights.pushDouble(layout.getHeight() / density);
 
-                if (includeWidths) {
+
+                if (mode == FlatHeightsMode.heightsWithHasEllipsisOnAndroid || mode == FlatHeightsMode.sizes) {
                     final int lineCount = layout.getLineCount();
-                    float measuredWidth = 0;
-                    for (int i = 0; i < lineCount; i++) {
-                        measuredWidth = Math.max(measuredWidth, layout.getLineMax(i));
+                    boolean lastLineHasEllipsis = layout.getEllipsisCount(lineCount - 1) > 0;
+                    hasEllipsisOnAndroid.pushBoolean(lastLineHasEllipsis);
+
+                    if (mode == FlatHeightsMode.sizes) {
+                        float measuredWidth = 0;
+                        for (int i = 0; i < lineCount; i++) {
+                            measuredWidth = Math.max(measuredWidth, layout.getLineMax(i));
+                        }
+                        widths.pushDouble(measuredWidth / density);
                     }
-                    widths.pushDouble(measuredWidth / density);
                 }
             }
 
-            if (includeWidths) {
-                final WritableMap output = Arguments.createMap();
-                output.putArray("widths", widths);
-                output.putArray("heights", heights);
-                promise.resolve(output);
-            } else {
-                promise.resolve(heights);
+            switch (mode) {
+                case sizes: {
+                    final WritableMap output = Arguments.createMap();
+                    output.putArray("widths", widths);
+                    output.putArray("heights", heights);
+                    output.putArray("hasEllipsisOnAndroid", hasEllipsisOnAndroid);
```

Comment:
Just checking, but might it make more sense to not have a separate array as the output for each of these, but instead have a single object like:

```ts
{
  width: number;
  height: number;
  isTextTruncatedOnAndroid?: boolean;
}
```